renderdoc/api/replay/shader_types.h

```diff
@@ -518,6 +518,34 @@ component-by-component to greatly simplify algorithms at the expense of a small
 space.
 )");
   rdcarray<DebugVariableReference> variables;
+
+  DOCUMENT(R"(Utility function for setting a bindpoint reference.
+
+See :class:`ShaderBindpointMapping` for the details of how bindpoints are interpreted. The type of
+binding is given by the :data:`type` member.
+
+:param int bindset: The bind set.
+:param int bind: The bind itself.
+:param int arrayIndex: The array index, if the bind is an array. If it isn't an array this should be
+  set to 0.
+)");
+  inline void SetBinding(int32_t bindset, int32_t bind, uint32_t arrayIndex)
+  {
+    rows = bindset;
+    columns = bind;
+    offset = arrayIndex;
+  }
+
+  DOCUMENT(R"(Utility function for getting the bindpoint referenced by this variable.
+
+.. note::
+
+  The return value is undefined if this variable is not a binding reference.
+
+:return: A :class:`BindpointIndex` with the binding referenced.
+:rtype: BindpointIndex
+)");
+  inline BindpointIndex GetBinding() const { return BindpointIndex(rows, columns, offset); }
```

Comment:
`SourceVariableMapping` shouldn't contain any direct data like this. It should only be used to map a high-level variable (if it exists) to a debug variable, potentially adding type information that's not present in the underlying variable. Thenm the debug variable itself can then contain any contents such as a binding for a resource register. That way this will still work as-is even if no high-level variable mapping information is available.

Reply:
Switched to storing the resource accesses in a `ShaderVariableChange`